tiledb/sm/c_api/tiledb.cc

```diff
@@ -4961,6 +4962,23 @@ int32_t tiledb_fragment_info_get_fragment_name(
   return TILEDB_OK;
 }
 
+int32_t tiledb_fragment_info_get_fragment_name_v2(
+    tiledb_ctx_t* ctx,
+    tiledb_fragment_info_t* fragment_info,
+    uint32_t fid,
+    tiledb_string_t** name) {
+  if (sanity_check(ctx) == TILEDB_ERR ||
+      sanity_check(ctx, fragment_info) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  const std::string& nameStr =
```

Comment:
As a rule you don't want to hold a reference when the function returns a value. It can work because of lifetime extension for temporaries bound to references, but it's a good habit to rely on that only when you really need to, not out of mere convenience.
In this case we don't even need a temporary variable. It would be better to inline the call.